st2api/st2api/controllers/v1/actionviews.py

```diff
@@ -98,32 +95,36 @@ class OverviewController(resource.ContentPackResourceController):
 
     include_reference = True
 
-    @jsexpose(arg_types=[str])
     def get_one(self, ref_or_id):
         """
             List action by id.
 
             Handle:
                 GET /actions/views/overview/1
         """
-        action_api = super(OverviewController, self)._get_one(ref_or_id)
-        return self._transform_action_api(action_api)
+        resp = super(OverviewController, self)._get_one(ref_or_id)
+        action_api = resp.json
+        result = self._transform_action_api(action_api)
+        resp.json = result
+        return resp
 
-    @jsexpose(arg_types=[str])
     def get_all(self, **kwargs):
         """
             List all actions.
 
             Handles requests:
                 GET /actions/views/overview
         """
-        action_apis = super(OverviewController, self)._get_all(**kwargs)
-        return map(self._transform_action_api, action_apis)
+        resp = super(OverviewController, self)._get_all(**kwargs)
+        action_apis = resp.json
+        result = map(self._transform_action_api, action_apis)
+        resp.json = result
+        return resp
 
     @staticmethod
     def _transform_action_api(action_api):
-        action_id = action_api.id
-        action_api.parameters = ParametersViewController._get_one(action_id).get('parameters')
+        action_id = action_api['id']
```

Comment:
I noticed this in other places as well - does this need to be a dict or it can be an object (aka specific class instance)?

I would much more prefer it to be an object - this way it's much easier to do static analysis and similar things (IDE auto-completion, etc.) and see what attributes are available.

I'm really not a fan of dicts, especially when dealing with objects like our API model which have a fixed limited set of possible attributes.

"Magical" dicts make programmer's life much harder and also make it easier to introduce bugs which would otherwise be caught if we used an object or similar.

I didn't dig in too much yet, but I assume that's swaggers thing (please correct me if I'm wrong) and I don't like it. I like what protobufs, thrift and other similar frameworks do much more - "definitions" are compiled and models are automatically generated from the definitions. This way it's same as any other static class definition in the code. It involves no magic, it makes static analysis and other things possible.